### ISSUE TITLE
Limit adding/removing Infra Servers and organizations to Editors

### ISSUE THREAD

**Issue by vsingh-msys** [chef/automate#3519]
## User Story

```
##### IAM
> - [ ] Limit adding/removing Infra Servers and organizations to Editors
- [x] Allow associating an organization to one or more projects
- [ ] Limit viewing policy (cookbooks, roles, etc) if the user lacks access (including filtered by project)


```

#1544 also required to implied policies for adding/removing Infra Servers & Organizations to `Editors` only access.

In order to do that need to modify the default `editor` role to append

`infra:infraServers:*,` but seems a bit late to raise the concern as related works in force upgrade V2 in PR https://github.com/chef/automate/pull/2783

At present, it has only `Administrators` only access as discussed it subject to change and needs to allow editors to manage the edit and delete operations.

**Comment by bcmdarroch** [comment]
Auth team had a conversation with @btm this morning that helped us decide what permission updates we need:

Permission behavior we want:
- Creating, updating, and removing chef servers and orgs should be reserved for admins only.
- Viewing data from those chef servers should be available to roles with lower privileges, Viewers and Editors.

To make this happen, we'll need to make the following changes:
- update the actions in [infra_proxy.proto](https://github.com/chef/automate/blob/master/api/external/infra_proxy/infra_proxy.proto) so that all GET APIs (GetCookbooks, GetEnvironment, GetOrgs, etc) have the action `infra:infraServers:get`. We can assume that if a user has access to Get a Chef Server, then they should be able to Get any of its sub-resources. This will simplify things for our customers and our Chef-managed policies.
- a special note on [chef server orgs](https://github.com/chef/automate/blob/master/api/external/infra_proxy/infra_proxy.proto#L72-L103): based on how @btm described the relationship between chef servers and orgs, I think it makes sense to merge server org permissions with the top-level chef server permissions. This is how our project rules work: creating a rule for a project falls under the same action as updating a project.
So instead of this format `infra:infraServerOrg:action`, we'd use these actions:
```
 rpc GetOrgs (infra_proxy.request.GetOrgs) returns (infra_proxy.response.GetOrgs)
    option (chef.automate.api.iam.policy).action = "infra:infraServers:get

  rpc GetOrg (infra_proxy.request.GetOrg) returns (infra_proxy.response.GetOrg)
    option (chef.automate.api.iam.policy).action = "infra:infraServers:get";

  rpc GetOrgByName (infra_proxy.request.GetOrgByName) returns (infra_proxy.response.GetOrg)
    option (chef.automate.api.iam.policy).action = "infra:infraServers:get";

  rpc CreateOrg (infra_proxy.request.CreateOrg) returns (infra_proxy.response.CreateOrg)
    option (chef.automate.api.iam.policy).action = "infra:infraServers:update";

  rpc UpdateOrg (infra_proxy.request.UpdateOrg) returns (infra_proxy.response.UpdateOrg)
    option (chef.automate.api.iam.policy).action = "infra:infraServers:update";

  rpc DeleteOrg (infra_proxy.request.DeleteOrg) returns (infra_proxy.response.DeleteOrg)
    option (chef.automate.api.iam.policy).action = "infra:infraServers:update";
```
- add an authz-service db migration that adds `infra:infraServers:get` as an action to both the Viewer and Editors roles

**Comment by vsingh-msys** [comment]
Thankyou @bcmdarroch for the summarized detail.

**Comment by vsingh-msys** [comment]
Fixed #3519 via #3550
